pkg/cmd/variable/get/get.go

```diff
@@ -115,14 +116,17 @@ func getRun(opts *GetOptions) error {
 
 	host, _ := cfg.Authentication().DefaultHost()
 
-	response := &Variable{}
+	var response getVariableResponse
+	if err = client.REST(host, "GET", path, nil, &response); err != nil {
+		var httpErr api.HTTPError
+		if errors.As(err, &httpErr) && httpErr.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("variable %s was not found", opts.VariableName)
+		}
 
-	err = client.REST(host, "GET", path, nil, &response)
-	if err != nil {
 		return fmt.Errorf("failed to get variable %s: %w", opts.VariableName, err)
 	}
 
-	fmt.Fprintf(opts.IO.Out, "%s", response.Value)
+	fmt.Fprintf(opts.IO.Out, "%s\n", response.Value)
```

Comment:
@williammartin I think adding the newline could be a problem in scripting.

When doing stuff like
```sh
FOO=$(gh variable get MY_VAR)
```

Now the trailing newline would be part of the variable. Of course, I could trim the trailing newline, but it's inconvenient and not obvious.

It's nice when the variable is displayed in the terminal (but if the variable actually had a newline already, you would get an extra).

A solution for increased readability in the terminal could be to add a newline but add it to stderr, if and only if the variable already lacks a trailing newline and both stdout and stderr are to the terminal.
```go
fmt.Fprintf(opts.IO.Out, "%s", response.Value)

if !strings.HasSuffix(response.Value, "\n") && opts.IO.IsStdoutTTY() && opts.IO.IsStderrTTY() {
	fmt.Fprint(opts.IO.ErrOut, "\n")
}
```

Reply:
Well...this sent me on a bit of a journey. Firstly, I appreciate your comment and sorry that I didn't recognise there was more to discuss here.

### Command Substitution

I don't believe that your first example is an issue because command substitution (`$()`) removes trailing line feed characters:

```bash
➜ export FOO="---\n\n\n"
➜ echo $FOO
---



➜ echo "$(echo "$FOO")"
---
```

Thus, our addition of a `\n` in the `fmt.Printf` should have no impact on `FOO=$(gh variable get MY_VAR)`. This is consistent with other `gh` commands when not hooked up to a TTY but printing info e.g. `repo create` showing the newly created repo URL. 

This also makes sense from a UX point of view as well because in CI or scripts we wouldn't want users to have to add newlines everywhere in order to avoid text appearing concatenated together with no spacing.

### However....

It did get me wondering about your other comment relating to variables ending with newlines. If command substitution strips trailing Line Feed chars then that will be problematic for variables that have trailing new lines:

```
~/workspace/cli/bin/gh variable set Triage9106 --body "foobar

"
foobar


✓ Updated variable Triage9106 for cli/cli
➜  ~/workspace/cli/bin/gh variable get Triage9106
foobar


➜ echo "$(~/workspace/cli/bin/gh variable get Triage9106)"
foobar
```

So this leaves us in an interesting situation because for variables ending in newlines, command substitution is a bit of a nightmare. There does appear to be some [workarounds](https://stackoverflow.com/a/15184414) however in that case it's probably not possible for a user to distinguish between their own newlines and the one added by the CLI.

However, I'm inclined to say that in this particular case, we document it and place the responsibility on the user to strip our trailing newline before processing the string because the alternative and much more common case seems quite annoying (strings being concatenated together and it working differently from other commands).

### Finally

I'm inclined to say there is a bug in both `secret set` and `variable set` when reading the variable value from stdin. It looks like in https://github.com/cli/cli/pull/5086 we decided to trim carriage return and line feed chars from the end because the shell adds **one** but the implementation actually strips all trailing CR/LF chars:

```
➜ echo $FOO | ~/workspace/cli/bin/gh variable set Triage9106
---
✓ Updated variable Triage9106 for cli/cli
➜ ~/workspace/cli/bin/gh variable get Triage9106
---
```

---

Let me know your thoughts, cheers!

Reply:
Oh. I didn't know command substitution removes trailing newlines.

It might just have been me being too careful about not loosing my scripting capabilities.

So adding the newline seems perfectly inline with what would normally be expected. I'm not sure, removing the newlines in command substitution was the world's best idea (looking at the workarounds, among others) but that should probably have been handled 45 years ago by someone not me or you :smile: